cpp/src/arrow/compute/kernels/aggregate_basic.cc

```diff
@@ -363,11 +381,13 @@ struct BooleanAllImpl : public ScalarAggregator {
     const auto& other = checked_cast<const BooleanAllImpl&>(src);
     this->all &= other.all;
     this->has_nulls |= other.has_nulls;
+    this->count += other.count;
     return Status::OK();
   }
 
   Status Finalize(KernelContext*, Datum* out) override {
-    if (!options.skip_nulls && this->all && this->has_nulls) {
+    if ((!options.skip_nulls && this->all && this->has_nulls) ||
+        this->count < options.min_count) {
```

Comment:
Same question here: if the input is `[false, true, null]` and `skip_nulls` is false, then the result is `false` rather than `null`?

Reply:
Yes, and this matches base R/dplyr's behavior:

```r
> all(c(FALSE, TRUE, NA), na.rm = FALSE)
[1] FALSE
```